src/main/java/org/fogbowcloud/manager/core/plugins/compute/openstack/OpenStackNovaV2ComputePlugin.java

```diff
@@ -0,0 +1,438 @@
+        LOGGER.debug("Generating JSON to send as the body of instance POST request...");
+
+        JSONObject server = new JSONObject();
+        server.put(NAME_JSON_FIELD, FOGBOW_INSTANCE_NAME + UUID.randomUUID().toString());
+        server.put(IMAGE_JSON_FIELD, imageRef);
+        server.put(FLAVOR_JSON_FIELD, flavorRef);
+
+        if (userdata != null) {
+            server.put(USER_DATA_JSON_FIELD, userdata);
+        }
+
+        if (networkId != null && !networkId.isEmpty()) {
+            List<JSONObject> nets = new ArrayList<>();
+            JSONObject net = new JSONObject();
+            net.put(UUID_JSON_FIELD, networkId);
+            nets.add(net);
+            server.put(NETWORK_JSON_FIELD, nets);
+        }
+
+        if (keyName != null && !keyName.isEmpty()) {
+            server.put(KEY_JSON_FIELD, keyName);
+        }
+
+        JSONObject root = new JSONObject();
+        root.put(SERVER_JSON_FIELD, server);
+
+        return root;
+    }
+
+    private void checkStatusResponse(HttpResponse response, String message) throws RequestException {
+        if (response.getStatusLine().getStatusCode() == HttpStatus.SC_UNAUTHORIZED) {
+            throw new RequestException(ErrorType.UNAUTHORIZED, ResponseConstants.UNAUTHORIZED);
+        } else if (response.getStatusLine().getStatusCode() == HttpStatus.SC_NOT_FOUND) {
+            throw new RequestException(ErrorType.NOT_FOUND, ResponseConstants.NOT_FOUND);
+        } else if (response.getStatusLine().getStatusCode() == HttpStatus.SC_BAD_REQUEST) {
+            throw new RequestException(ErrorType.BAD_REQUEST, message);
+        } else if (response.getStatusLine().getStatusCode() == HttpStatus.SC_REQUEST_TOO_LONG
+                || response.getStatusLine().getStatusCode() == HttpStatus.SC_FORBIDDEN) {
+            if (message.contains(ResponseConstants.QUOTA_EXCEEDED_FOR_INSTANCES)) {
+                throw new RequestException(ErrorType.QUOTA_EXCEEDED,
+                        ResponseConstants.QUOTA_EXCEEDED_FOR_INSTANCES);
+            }
+            throw new RequestException(ErrorType.BAD_REQUEST, message);
+        } else if ((response.getStatusLine().getStatusCode() == HttpStatus.SC_INTERNAL_SERVER_ERROR) &&
+                (message.contains(NO_VALID_HOST_WAS_FOUND))) {
+            throw new RequestException(ErrorType.NO_VALID_HOST_FOUND, ResponseConstants.NO_VALID_HOST_FOUND);
+        } else if (response.getStatusLine().getStatusCode() > 204) {
+            throw new RequestException(ErrorType.BAD_REQUEST,
+                    "Status code: " + response.getStatusLine().toString() + " | Message:" + message);
+        }
+    }
+
+    protected Flavor updateAndFindSmallestFlavor(ComputeOrder computeOrder) {
+        updateFlavors(computeOrder.getLocalToken());
+
+        return findSmallestFlavor(computeOrder);
+    }
+
+    private Flavor findSmallestFlavor(ComputeOrder computeOrder) {
+        List<Flavor> listFlavor = new ArrayList<>();
```

Comment:
This method can be simplified using a TreeSet DS instead of a List and getting the smallest flavor that matches with the computeOrder by the [ceiling](https://docs.oracle.com/javase/7/docs/api/java/util/TreeSet.html) method (binary search).

Therefore, this method will be used by others computePlugin implementations, you could put this method in a utility class.